tests/test_optimizer.py

```diff
@@ -814,3 +818,72 @@ def test_query_annotate_with_callable(db, gql_client: GraphQLTestClient):
                     asserts_errors=False,
                 )
                 assert res.errors
+
+
+@pytest.mark.django_db(transaction=True)
+def test_user_query_with_prefetch():
+    @strawberry_django.type(
+        Project,
+    )
+    class ProjectTypeWithPrefetch:
+        @strawberry_django.field(
+            prefetch_related=[
+                Prefetch(
```

Comment:
I can see the issue here =P

I usually define `Prefetch` objects as callables (i.e. `prefetch_related=[lambda info: Prefetch(...)]`), meaning each call will have its own prefetch, so I didn't notice the possibility of this issue

Nice catch!

Reply:
He, didn't think of that, I just stumbled upon it by using the same syntax from the test, defining the `Prefetch` objects directly.